Mathlib/AlgebraicTopology/Nerve.lean

```diff
@@ -49,4 +74,387 @@ lemma δ₀_eq {x : nerve C _[n + 1]} : (nerve C).δ (0 : Fin (n + 2)) x = x.δ
+
+/-- An object `j : StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op` defines a morphism
+`Fin (jlen+1) -> Fin(n+1)`. This calculates the image of `i : Fin(jlen+1)`;
+we might think of this as j(i). -/
+private
+def strArr.homEv {n}
+    (j : StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op)
+    (i : Fin ((unop ((Truncated.inclusion (n := 2)).op.obj
+      ((StructuredArrow.proj (op [n]) (Truncated.inclusion (n := 2)).op).obj j))).len + 1)) :
+    Fin (n + 1) := (SimplexCategory.Hom.toOrderHom j.hom.unop) i
+
+/-- This is the unique arrow in `StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op` from
+j to pt' of the j(i) calculated above. This is used to prove that ran.lift defines a factorization
+on objects.-/
+private
+def fact.obj.arr {n}
+    (j : StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op)
+    (i : Fin ((unop ((Truncated.inclusion (n := 2)).op.obj
+      ((StructuredArrow.proj (op [n]) (Truncated.inclusion (n := 2)).op).obj j))).len + 1)) :
+      j ⟶ (pt' (strArr.homEv j i)) :=
+  StructuredArrow.homMk (.op (SimplexCategory.const _ _ i)) <| by
+    apply Quiver.Hom.unop_inj
+    ext z; revert z; intro | 0 => rfl
+
+/-- An object `j : StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op` defines a morphism
+`Fin (jlen+1) -> Fin(n+1)`. This calculates the image of i.succ : Fin(jlen+1); we might think of
+this as j(i.succ). -/
+private
+def strArr.homEvSucc {n}
+    (j : StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op)
+    (i : Fin (unop j.right).1.len) :
+    Fin (n + 1) := (SimplexCategory.Hom.toOrderHom j.hom.unop) i.succ
+
+/-- The unique arrow (strArr.homEv j i.castSucc) ⟶ (strArr.homEvSucc j i) in Fin(n+1). -/
+private
+def strArr.homEv.map {n}
+    (j : StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op)
+    (i : Fin (unop j.right).1.len) :
+    strArr.homEv j i.castSucc ⟶ strArr.homEvSucc j i :=
+  (Monotone.functor (j.hom.unop.toOrderHom).monotone).map (Fin.hom_succ i)
+
+/-- This is the unique arrow in `StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op` from j
+to ar' of the map just
+constructed. This is used to prove that ran.lift defines a factorization on maps.-/
+private
+def fact.map.arr {n}
+    (j : StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op)
+    (i : Fin (unop j.right).1.len) : j ⟶ ar' (strArr.homEv.map j i) := by
+  fapply StructuredArrow.homMk
+  · exact .op (mkOfSucc i : [1] ⟶ [(unop j.right).1.len])
+  · apply Quiver.Hom.unop_inj
+    ext z; revert z
+    intro
+    | 0 => rfl
+    | 1 => rfl
+
+/-- For each natural number `n`, `nerveRightExtension.coneAt C n` defines a cone with summit
+`nerve C _[n]` over the diagram
+`(StructuredArrow.proj (op [n]) (Truncated.inclusion (n := 2)).op ⋙ nerveFunctor₂.obj C)`
+indexed by the category `StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op.` This proves
+that this cone is a limit cone.-/
+noncomputable def isPointwiseRightKanExtensionAt (C : Cat) (n : ℕ) :
```

Comment:
It seems that this is the critical part. I would suggest introducing paths in simplicial sets as follows, and showing that if a simplicial set `X` is such that the type of `n`-simplices maps bijectively to `X.Path n` for all `n`, then the associated right extension is a pointwise right Kan extension: 
```lean
namespace SSet

open CategoryTheory

variable (X : SSet.{u})

@[ext]
structure Path (n : ℕ) where
  vertex (i : Fin (n + 1)) : X _[0]
  arrow (i : Fin n) : X _[1]
  arrow_src (i : Fin n) : X.δ 1 (arrow i) = vertex i.castSucc
  arrow_tgt (i : Fin n) : X.δ 0 (arrow i) = vertex i.succ

@[simps]
def toPath (n : ℕ) (Δ : X _[n]) :
    X.Path n where
  vertex i := X.map (SimplexCategory.const [0] [n] i).op Δ
  arrow i := X.map (SimplexCategory.mkOfLe _ _ i.castSucc_le_succ).op Δ
  arrow_src i := by
    dsimp [SimplicialObject.δ]
    simp only [← FunctorToTypes.map_comp_apply, ← op_comp]
    apply congr_fun
    congr 2
    ext i
    fin_cases i
    rfl
  arrow_tgt i := by
    dsimp [SimplicialObject.δ]
    simp only [← FunctorToTypes.map_comp_apply, ← op_comp]
    apply congr_fun
    congr 2
    ext i
    fin_cases i
    rfl

lemma toPath_nerve_bijective (C : Type u) [Category.{v} C] (n : ℕ) :
    Function.Bijective ((nerve C).toPath n) := by
  constructor
  · intro Δ Δ' h
    exact ComposableArrows.ext
      (fun i ↦ Functor.congr_obj (congr_fun (congr_arg Path.vertex h) i) 0)
      (fun i hi ↦ congr_hom (congr_fun (congr_arg Path.arrow h) ⟨i, hi⟩) (show 0 ⟶ 1 by tauto))
  · intro F
    exact ⟨ComposableArrows.mkOfObjOfMapSucc (fun i ↦ (F.vertex i).obj 0)
      (fun i ↦ eqToHom (Functor.congr_obj (F.arrow_src i).symm 0) ≫
        (F.arrow i).map' 0 1 ≫ eqToHom (Functor.congr_obj (F.arrow_tgt i) 0)), ?_⟩
    ext i
    · exact ComposableArrows.ext₀ rfl
    · refine ComposableArrows.ext₁ ?_ ?_ ?_
      · exact Functor.congr_obj (F.arrow_src i).symm 0
      · exact Functor.congr_obj (F.arrow_tgt i).symm 0
      · dsimp
        apply ComposableArrows.mkOfObjOfMapSucc_map_succ

@[simps!]
def rightExtensionInclusion (n : ℕ) :
    RightExtension (Truncated.inclusion (n := n)).op
      (Functor.op Truncated.inclusion ⋙ X) :=
  RightExtension.mk _ (𝟙 _)

noncomputable def rightExtensionInclusion₂IsPointwiseRightKanExtensionAt
    (hX : ∀ (n : ℕ), Function.Bijective (X.toPath n)) (n : ℕ) :
    (X.rightExtensionInclusion 2).IsPointwiseRightKanExtensionAt ⟨[n]⟩ := sorry

end SSet
```
(This is related to a criteria which identifies which simplicial sets are the nerve of a category.)

Reply:
(The ingredients in the proof should be roughly the same. Hopefully, it may be a little bit less painful.)

Reply:
Thanks for the suggestion. Less pain sounds good! I'll get to work.